src/m_alloc.cpp

```diff
@@ -52,14 +54,14 @@
 #endif
 #if defined(__APPLE__)
 #define _msize(p)				malloc_size(p)
-#elif defined(__sun)
+#elif __solaris__ || defined(__OpenBSD__)
```

Comment:
What is with changing __sun to __solaris__ here? I am very weary of approving the PR without having an actual Solaris system to test on so that I know that these defines still work the same.